include/nbl/builtin/hlsl/ieee754.hlsl

```diff
@@ -142,13 +142,47 @@ NBL_CONSTEXPR_INLINE_FUNC FloatingPoint copySign(FloatingPoint to, FloatingPoint
 	return bit_cast<FloatingPoint>(toAsUint | extractSignPreserveBitPattern(from));
 }
 
-template <typename FloatingPoint NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<FloatingPoint>)
-NBL_CONSTEXPR_INLINE_FUNC FloatingPoint flipSign(FloatingPoint val, bool flip = true)
+namespace impl
+{
+template <typename T NBL_STRUCT_CONSTRAINABLE>
+struct flipSign_helper;
+
+template <typename FloatingPoint>
+NBL_PARTIAL_REQ_TOP(concepts::FloatingPointLikeScalar<FloatingPoint>)
+struct flipSign_helper<FloatingPoint NBL_PARTIAL_REQ_BOT(concepts::FloatingPointLikeScalar<FloatingPoint>) >
+{
+	static FloatingPoint __call(FloatingPoint val, bool flip = true)
```

Comment:
`flip` needs be be `U` which is allowed to be scalar when `FloatingPoint` is scalar or vectorial of same dimension otherwise

Reply:
This one is a partial spec for handling scalar types. But I'll have to add one for boolean vector  `flip`.